File: p3.java
// Time Complexity : O(n)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No

// Your code here along with comments explaining your approach

class Solution {
    public int maxArea(int[] height) {
        // Declaring variable max which is the output that we will return. Declaring two
        // pinters low and high
        int max = 0;
        int l = 0;
        int h = height.length - 1;

        // Edge case if there is no line or only one line
        if (h < 1) {
            return max;
        }

        // Loop will run till low is less than high. Incremeent low or high depending on
        // which line is lower. Also calculate max while doing that
        while (l < h) {
            max = Math.max(max, (Math.min(height[l], height[h])) * (h - l));
            if (height[l] < height[h]) {
                l++;
            } else {
                h--;
            }
        }
        return max;
    }
}

File: P3.py
# -*- coding: utf-8 -*-
"""
Created on Mon Apr 12 11:38:15 2021

@author: jgdch
"""

#time Complexity:O(N), N is length of input
#Space complexity: O(1)
#Passed all leetcode testcases
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l=0
        r=len(height)-1
        area=0
        while(l<r):
            area=max(area,min(height[l],height[r])*(r-l))
            if(height[l]<=height[r]):
                l+=1
            else:
                r-=1
        return area
            